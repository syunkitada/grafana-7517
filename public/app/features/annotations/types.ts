import { PanelData, DataFrame, AnnotationEvent, TimeRange } from '@grafana/data';
import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * All the data flattened to a single frame
   */
  frame?: DataFrame;

  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}
