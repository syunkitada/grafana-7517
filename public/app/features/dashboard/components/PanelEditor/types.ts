import { PanelModel } from '../../state/PanelModel';

export enum DisplayMode {
  Fill = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Fill, label: 'Fill', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];

export enum EditorTab {
  Query = 'query',
  Alerts = 'alerts',
  Transform = 'xform',
}

export const allTabs = [
  { tab: EditorTab.Query, label: 'Query', show: (panel: PanelModel) => true },
  { tab: EditorTab.Alerts, label: 'Alerts', show: (panel: PanelModel) => true },
  { tab: EditorTab.Transform, label: 'Transform', show: (panel: PanelModel) => true },
];
