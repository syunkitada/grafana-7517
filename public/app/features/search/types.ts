export enum DashboardSearchItemType {
  DashDB = 'dash-db',
  DashHome = 'dash-home',
  DashFolder = 'dash-folder',
}

export interface DashboardSection {
  id: number;
  uid?: string;
  title: string;
  expanded?: boolean;
  url: string;
  icon: string;
  score: number;
  hideHeader?: boolean;
  checked?: boolean;
  items: DashboardSectionItem[];
  toggle?: (section: DashboardSection) => Promise<DashboardSection>;
  selected?: boolean;
  type: DashboardSearchItemType;
}

export interface DashboardSectionItem {
  checked?: boolean;
  folderId?: number;
  folderTitle?: string;
  folderUid?: string;
  folderUrl?: string;
  id: number;
  isStarred: boolean;
  selected?: boolean;
  tags: string[];
  title: string;
  type: DashboardSearchItemType;
  uid: string;
  uri: string;
  url: string;
}

export interface DashboardTag {
  term: string;
  count: number;
}

export interface DashboardQuery {
  query: string;
  mode: string;
  tag: string[];
  starred: boolean;
  skipRecent: boolean;
  skipStarred: boolean;
  folderIds: number[];
}

export interface SectionsState {
  sections: DashboardSection[];
  allChecked: boolean;
  dashboardTags: DashboardTag[];
}

export type ItemClickWithEvent = (item: DashboardSectionItem | DashboardSection, event: any) => void;

export type SearchAction = {
  type: string;
  payload?: any;
};

export interface OpenSearchParams {
  query?: string;
}
