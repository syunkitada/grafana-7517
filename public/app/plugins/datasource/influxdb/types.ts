import { DataQuery, DataSourceJsonData } from '@grafana/data';

export interface InfluxOptions extends DataSourceJsonData {
  timeInterval: string;
  httpMode: string;

  // Influx 2.0
  enableFlux?: boolean;
  organization?: string;
  defaultBucket?: string;
  maxSeries?: number;
}

export interface InfluxSecureJsonData {
  password?: string;
  token?: string;
}

export interface InfluxQueryPart {
  type: string;
  params?: string[];
  interval?: string;
}

export interface InfluxQueryTag {
  key: string;
  operator?: string;
  condition?: string;
  value: string;
}

export enum InfluxQueryType {
  Classic = 'Classic', // IFQL query builder
  InfluxQL = 'InfluxQL', // raw ifql
  Flux = 'Flux',
}

export interface InfluxQuery extends DataQuery {
  queryType?: InfluxQueryType;
  policy?: string;
  measurement?: string;
  resultFormat?: 'time_series' | 'table';
  orderByTime?: string;
  tags?: InfluxQueryTag[];
  groupBy?: InfluxQueryPart[];
  select?: InfluxQueryPart[][];
  limit?: string;
  slimit?: string;
  tz?: string;
  fill?: string;
  rawQuery?: boolean; // deprecated (use raw InfluxQL)
  query?: string;
}
