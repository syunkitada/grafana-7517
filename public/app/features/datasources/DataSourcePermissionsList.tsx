import React, { PureComponent } from 'react';
import { DataSourcePermission } from '../../types';
import { dataSourceAclLevels, DataSourcePermissionLevel } from '../../types/acl';
import DescriptionPicker from '../../core/components/Picker/DescriptionPicker';

export interface Props {
  items: DataSourcePermission[];
  onRemoveItem: (item) => void;
}

export class DataSourcePermissionsList extends PureComponent<Props> {
  renderAvatar(item) {
    if (item.teamId) {
      return <img className="filter-table__avatar" src={item.teamAvatarUrl} />;
    } else if (item.userId) {
      return <img className="filter-table__avatar" src={item.userAvatarUrl} />;
    }

    return <i style={{ width: '25px', height: '25px' }} className="gicon gicon-viewer" />;
  }

  renderDescription(item) {
    if (item.userId) {
      return [
        <span key="name">{item.userLogin} </span>,
        <span key="description" className="filter-table__weak-italic">
          (User)
        </span>,
      ];
    }
    if (item.teamId) {
      return [
        <span key="name">{item.team} </span>,
        <span key="description" className="filter-table__weak-italic">
          (Team)
        </span>,
      ];
    }
    return <span className="filter-table__weak-italic">(Role)</span>;
  }

  render() {
    const { items } = this.props;
    const permissionLevels = dataSourceAclLevels;
    permissionLevels.push({ value: DataSourcePermissionLevel.Admin, label: 'Admin', description: '' });

    return (
      <table className="filter-table gf-form-group">
        <tbody>
          <tr className="gf-form-disabled">
            <td style={{ width: '1%' }}>
              <i style={{ width: '25px', height: '25px' }} className="gicon gicon-shield" />
            </td>
            <td style={{ width: '90%' }}>
              Admin
              <span className="filter-table__weak-italic"> (Role)</span>
            </td>
            <td />
            <td className="query-keyword">Can</td>
            <td>
              <div className="gf-form">
                <DescriptionPicker
                  optionsWithDesc={permissionLevels}
                  onSelected={() => {}}
                  value={2}
                  disabled={true}
                  className={'gf-form-input--form-dropdown-right'}
                />
              </div>
            </td>
            <td>
              <button className="btn btn-inverse btn-small">
                <i className="fa fa-lock" />
              </button>
            </td>
          </tr>
          {items.map((item, index) => {
            return (
              <tr key={`${item.id}-${index}`}>
                <td style={{ width: '1%' }}>{this.renderAvatar(item)}</td>
                <td style={{ width: '90%' }}>{this.renderDescription(item)}</td>
                <td />
                <td className="query-keyword">Can</td>
                <td>
                  <div className="gf-form">
                    <DescriptionPicker
                      optionsWithDesc={permissionLevels}
                      onSelected={() => {}}
                      value={1}
                      disabled={true}
                      className={'gf-form-input--form-dropdown-right'}
                    />
                  </div>
                </td>
                <td>
                  <button className="btn btn-danger btn-small" onClick={() => this.props.onRemoveItem(item)}>
                    <i className="fa fa-remove" />
                  </button>
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    );
  }
}

export default DataSourcePermissionsList;
