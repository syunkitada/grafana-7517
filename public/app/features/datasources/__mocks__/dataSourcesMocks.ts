import { DataSource, DataSourcePermission } from 'app/types';

export const getMockDataSources = (amount: number): DataSource[] => {
  const dataSources = [];

  for (let i = 0; i <= amount; i++) {
    dataSources.push({
      access: '',
      basicAuth: false,
      database: `database-${i}`,
      id: i,
      isDefault: false,
      jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
      name: `dataSource-${i}`,
      orgId: 1,
      password: '',
      readOnly: false,
      type: 'cloudwatch',
      typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
      url: '',
      user: '',
    });
  }

  return dataSources;
};

export const getMockDataSource = (): DataSource => {
  return {
    access: '',
    basicAuth: false,
    basicAuthUser: '',
    basicAuthPassword: '',
    withCredentials: false,
    database: '',
    id: 13,
    isDefault: false,
    jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
    name: 'gdev-cloudwatch',
    orgId: 1,
    password: '',
    readOnly: false,
    type: 'cloudwatch',
    typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
    url: '',
    user: '',
  };
};

export const getMockDataSourcePermissionsUser = (): DataSourcePermission => {
  return {
    created: '2018-10-10T16:50:45+02:00',
    datasourceId: 1,
    id: 2,
    permission: 1,
    permissionName: 'Query',
    updated: '2018-10-10T16:50:45+02:00',
    userAvatarUrl: '/avatar/926aa85c6bcefa0b4deca3223f337ae1',
    userEmail: 'test@test.com',
    userId: 3,
    userLogin: 'testUser',
  };
};

export const getMockDataSourcePermissionsTeam = (): DataSourcePermission => {
  return {
    created: '2018-10-10T16:57:09+02:00',
    datasourceId: 1,
    id: 6,
    permission: 1,
    permissionName: 'Query',
    team: 'A-team',
    teamAvatarUrl: '/avatar/93c0801b955cbd443a8cfa91a401d7bc',
    teamId: 1,
    updated: '2018-10-10T16:57:09+02:00',
  };
};
