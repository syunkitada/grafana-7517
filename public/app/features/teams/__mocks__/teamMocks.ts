import { Team } from '../../../types';

export const getMultipleMockTeams = (numberOfTeams: number): Team[] => {
  let teams: Team[] = [];
  for (let i = 1; i <= numberOfTeams; i++) {
    teams.push({
      id: i,
      name: `test-${i}`,
      avatarUrl: 'some/url/',
      email: `test-${i}@test.com`,
      memberCount: i,
      search: '',
      members: [],
      groups: [],
    });
  }

  return teams;
};

export const getMockTeam = (): Team => {
  return {
    id: 1,
    name: 'test',
    avatarUrl: 'some/url/',
    email: 'test@test.com',
    memberCount: 1,
    search: '',
    members: [],
    groups: [],
  };
};
