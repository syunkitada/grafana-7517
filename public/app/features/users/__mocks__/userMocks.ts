export const getMockUsers = (amount: number) => {
  const users = [];

  for (let i = 0; i <= amount; i++) {
    users.push({
      avatarUrl: 'url/to/avatar',
      email: `user-${i}@test.com`,
      lastSeenAt: '2018-10-01',
      lastSeenAtAge: '',
      login: `user-${i}`,
      orgId: 1,
      role: 'Admin',
      userId: i,
    });
  }

  return users;
};

export const getMockUser = () => {
  return {
    avatarUrl: 'url/to/avatar',
    email: `user@test.com`,
    lastSeenAt: '2018-10-01',
    lastSeenAtAge: '',
    login: `user`,
    orgId: 1,
    role: 'Admin',
    userId: 2,
  };
};
