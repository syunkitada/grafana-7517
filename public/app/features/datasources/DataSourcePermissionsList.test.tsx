import React from 'react';
import { shallow } from 'enzyme';
import { DataSourcePermissionsList, Props } from './DataSourcePermissionsList';
import { DataSourcePermission } from '../../types';
import { getMockDataSourcePermissionsTeam, getMockDataSourcePermissionsUser } from './__mocks__/dataSourcesMocks';

const setup = (propOverrides?: object) => {
  const props: Props = {
    items: [] as DataSourcePermission[],
    onRemoveItem: jest.fn(),
  };

  Object.assign(props, propOverrides);

  return shallow(<DataSourcePermissionsList {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render items', () => {
    const wrapper = setup({
      items: [getMockDataSourcePermissionsUser(), getMockDataSourcePermissionsTeam()],
    });

    expect(wrapper).toMatchSnapshot();
  });
});
