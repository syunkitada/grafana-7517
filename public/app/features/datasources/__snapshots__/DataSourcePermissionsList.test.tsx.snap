// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<table
  className="filter-table gf-form-group"
>
  <tbody>
    <tr
      className="gf-form-disabled"
    >
      <td
        style={
          Object {
            "width": "1%",
          }
        }
      >
        <i
          className="gicon gicon-shield"
          style={
            Object {
              "height": "25px",
              "width": "25px",
            }
          }
        />
      </td>
      <td
        style={
          Object {
            "width": "90%",
          }
        }
      >
        Admin
        <span
          className="filter-table__weak-italic"
        >
           (Role)
        </span>
      </td>
      <td />
      <td
        className="query-keyword"
      >
        Can
      </td>
      <td>
        <div
          className="gf-form"
        >
          <DescriptionPicker
            className="gf-form-input--form-dropdown-right"
            disabled={true}
            onSelected={[Function]}
            optionsWithDesc={
              Array [
                Object {
                  "description": "Can query data source.",
                  "label": "Query",
                  "value": 1,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
              ]
            }
            value={2}
          />
        </div>
      </td>
      <td>
        <button
          className="btn btn-inverse btn-small"
        >
          <i
            className="fa fa-lock"
          />
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Render should render items 1`] = `
<table
  className="filter-table gf-form-group"
>
  <tbody>
    <tr
      className="gf-form-disabled"
    >
      <td
        style={
          Object {
            "width": "1%",
          }
        }
      >
        <i
          className="gicon gicon-shield"
          style={
            Object {
              "height": "25px",
              "width": "25px",
            }
          }
        />
      </td>
      <td
        style={
          Object {
            "width": "90%",
          }
        }
      >
        Admin
        <span
          className="filter-table__weak-italic"
        >
           (Role)
        </span>
      </td>
      <td />
      <td
        className="query-keyword"
      >
        Can
      </td>
      <td>
        <div
          className="gf-form"
        >
          <DescriptionPicker
            className="gf-form-input--form-dropdown-right"
            disabled={true}
            onSelected={[Function]}
            optionsWithDesc={
              Array [
                Object {
                  "description": "Can query data source.",
                  "label": "Query",
                  "value": 1,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
              ]
            }
            value={2}
          />
        </div>
      </td>
      <td>
        <button
          className="btn btn-inverse btn-small"
        >
          <i
            className="fa fa-lock"
          />
        </button>
      </td>
    </tr>
    <tr
      key="2-0"
    >
      <td
        style={
          Object {
            "width": "1%",
          }
        }
      >
        <img
          className="filter-table__avatar"
          src="/avatar/926aa85c6bcefa0b4deca3223f337ae1"
        />
      </td>
      <td
        style={
          Object {
            "width": "90%",
          }
        }
      >
        <span
          key="name"
        >
          testUser
           
        </span>
        <span
          className="filter-table__weak-italic"
          key="description"
        >
          (User)
        </span>
      </td>
      <td />
      <td
        className="query-keyword"
      >
        Can
      </td>
      <td>
        <div
          className="gf-form"
        >
          <DescriptionPicker
            className="gf-form-input--form-dropdown-right"
            disabled={true}
            onSelected={[Function]}
            optionsWithDesc={
              Array [
                Object {
                  "description": "Can query data source.",
                  "label": "Query",
                  "value": 1,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
              ]
            }
            value={1}
          />
        </div>
      </td>
      <td>
        <button
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </button>
      </td>
    </tr>
    <tr
      key="6-1"
    >
      <td
        style={
          Object {
            "width": "1%",
          }
        }
      >
        <img
          className="filter-table__avatar"
          src="/avatar/93c0801b955cbd443a8cfa91a401d7bc"
        />
      </td>
      <td
        style={
          Object {
            "width": "90%",
          }
        }
      >
        <span
          key="name"
        >
          A-team
           
        </span>
        <span
          className="filter-table__weak-italic"
          key="description"
        >
          (Team)
        </span>
      </td>
      <td />
      <td
        className="query-keyword"
      >
        Can
      </td>
      <td>
        <div
          className="gf-form"
        >
          <DescriptionPicker
            className="gf-form-input--form-dropdown-right"
            disabled={true}
            onSelected={[Function]}
            optionsWithDesc={
              Array [
                Object {
                  "description": "Can query data source.",
                  "label": "Query",
                  "value": 1,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
                Object {
                  "description": "",
                  "label": "Admin",
                  "value": 2,
                },
              ]
            }
            value={1}
          />
        </div>
      </td>
      <td>
        <button
          className="btn btn-danger btn-small"
          onClick={[Function]}
        >
          <i
            className="fa fa-remove"
          />
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
